client/src/components/MultiJurisdictionAudit/Progress/ProgressMap.tsx

```diff
@@ -0,0 +1,367 @@
+import { feature } from 'topojson-client'
+// topojson-specification is defined in package.json but throwing linting error here
+// eslint-disable-next-line import/no-unresolved
+import { Topology } from 'topojson-specification'
+import { Feature } from 'geojson'
+import { Colors, Spinner } from '@blueprintjs/core'
+import labelValueStates from '../AASetup/Settings/states'
+import { JurisdictionRoundStatus, IJurisdiction } from '../useJurisdictions'
+import { FileProcessingStatus, IFileInfo } from '../useCSV'
+
+interface IProps {
+  stateName?: string | null
+  jurisdictions: IJurisdiction[]
+  isRoundStarted: boolean
+  auditType:
+    | 'BALLOT_POLLING'
+    | 'BATCH_COMPARISON'
+    | 'BALLOT_COMPARISON'
+    | 'HYBRID'
+}
+
+const MapWrapper = styled.div`
+  position: relative;
+`
+
+const SVGMap = styled.svg`
+  margin-bottom: 30px;
+  .outline {
+    fill: none;
+    stroke: ${Colors.BLACK};
+    stroke-width: 0.5px;
+  }
+  .mesh {
+    fill: none;
+    stroke-linejoin: round;
+  }
+  path {
+    fill: none;
+    stroke: none;
+    stroke-width: 0.5px;
+  }
+  #single-state {
+    stroke: ${Colors.BLACK};
+  }
+  .county {
+    cursor: pointer;
+    fill: ${Colors.WHITE};
+    stroke: ${Colors.BLACK};
+  }
+  .county:hover {
+    stroke-width: 2px;
+  }
+  .danger {
+    fill: ${Colors.RED3};
+  }
+  .success {
+    fill: ${Colors.GREEN4};
+  }
+  .progress {
+    fill: ${Colors.ORANGE4};
+  }
+  .gray {
+    fill: ${Colors.GRAY4};
+  }
+  .default {
+    fill: ${Colors.WHITE};
+  }
+`
+
+const Tooltip = styled.div`
+  display: none;
+  position: absolute;
+  opacity: 0.9;
+  border-radius: 3px;
+  background: ${Colors.BLACK};
+  padding: 5px;
+  color: ${Colors.WHITE};
+`
+
+const MapLabels = styled.div`
+  position: absolute;
+  top: 0;
+  z-index: 10;
+`
+
+const MapLabelsRow = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 5px;
+`
+
+const MapLabelsBoxes = styled.div`
+  display: inline-block;
+  margin-right: 10px;
+  width: 20px;
+  height: 20px;
+  &.danger {
+    background-color: ${Colors.RED3};
+  }
+  &.success {
+    background-color: ${Colors.GREEN4};
+  }
+  &.progress {
+    background-color: ${Colors.ORANGE4};
+  }
+  &.gray {
+    background-color: ${Colors.GRAY4};
+  }
+`
+
+const MapSpinner = styled(Spinner)`
+  position: absolute;
+  top: 50%;
+  right: 0;
+  left: 0;
+  margin: 0 auto;
+`
+
+const Map = ({
+  stateName,
+  jurisdictions,
+  isRoundStarted,
+  auditType,
+}: IProps) => {
+  const width = 960
+  const height = 500
+  const d3Container = useRef(null)
+  const tooltipContainer = useRef(null)
+
+  const [usCounties, setUSCounties] = useState<Feature[] | undefined>(undefined)
+  const [usStates, setUSStates] = useState<Feature[] | undefined>(undefined)
+  const [usState, setUSState] = useState<Feature | undefined>(undefined)
+  const [jsonData, setJsonData] = useState<Topology | undefined>(undefined)
+  const [isMapLoaded, setIsMapLoaded] = useState<boolean>(false)
```

Comment:
what are we waiting for when we show the spinner? does it take time to load the topology data, or to execute the d3 code to set up the map? my guess is that it's almost all waiting for loading the topology data. if so, it might be virtually equivalent to stop showing the spinner as soon as the topology data is loaded, which means we could eliminate the `isMapLoaded` state and just check whether the topology data is loaded